src/app/videos/videos.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+
+import { videoInfo } from '../types/videoInfo';
+import { viewCount } from '../types/viewCount';
+import { thumbNailUrl } from '../types/thumbNailUrl';
+
+import { SharedDataService } from '../shared-data.service';
+
+@Component({
+  selector: 'app-videos',
+  templateUrl: './videos.component.html',
+  styleUrls: ['./videos.component.scss'],
+})
+export class VideosComponent {
+  randomVideos: videoInfo[] = [];
+  search: boolean = false;
+  constructor(
+    private sharedDataService: SharedDataService,
+    private route: ActivatedRoute,
+    private location: Location
+  ) {}
+
+  /**
+   * get the data from the service and merge them all in randomVideos variable
+   * and the data type is videoInfo[] and the data depends on random or searchedWord
+   *
+   * @returns {void} This function does not return a value directly.
+   **/
+  ngOnInit() {
+    if (!this.location.path()) {
+      this.getRandomVideos();
+    } else {
+      this.getSearchedVideos();
+    }
+  }
+
+  getRandomVideos() {
+    this.search = false;
+    this.sharedDataService.getRandomVideos().subscribe((data) => {
+      this.randomVideos = data;
+      this.mergeData();
+    });
+  }
+
+  getSearchedVideos() {
+    this.route.queryParams.subscribe((params) => {
+      this.sharedDataService
+        .getSearchedVideos(params['word'])
+        .subscribe((data) => {
+          this.search = true;
+          this.randomVideos = data;
+          this.mergeData();
+        });
```

Comment:
if we could make this cleaner by moving callbacks function out of the subscribe as previous comment